tests/models/test_instrumented.py

```diff
@@ -827,3 +827,59 @@ def test_messages_to_otel_events_without_binary_content(document_content: Binary
             }
         ]
     )
+
+
+def test_messages_to_otel_events_without_prompts_and_completions(document_content: BinaryContent):
+    messages: list[ModelMessage] = [
+        ModelRequest(parts=[SystemPromptPart('system_prompt')]),
+        ModelResponse(parts=[TextPart('text1')]),
+        ModelRequest(parts=[UserPromptPart('user_prompt')]),
+        ModelResponse(parts=[TextPart('text2')]),
+        ModelRequest(parts=[ToolReturnPart('tool', 'tool_return_content', 'tool_call_1')]),
+        ModelRequest(parts=[RetryPromptPart('retry_prompt', tool_name='tool', tool_call_id='tool_call_2')]),
+        ModelRequest(parts=[UserPromptPart(content=['user_prompt6', document_content])]),
```

Comment:
We definitely want it to be visible that there was a document or something there, even if we don't see the contents. Sorry, I know I said the `content` key should be omitted entirely, but I was only thinking about the case where it was just a string. In fact when there's an array containing both text and other stuff like images, we should be able to see the kinds of everything, so text should be replaced by `{kind:text}`.

Reply:
Yes, @Kludex what do you think?